src/lib/core/mp_interval.c

```diff
@@ -99,13 +99,34 @@ interval_pack(char *data, const struct interval *itv)
 }
 
 struct interval *
-interval_unpack(const char **data, struct interval *itv)
+interval_unpack(const char **data, uint32_t len, struct interval *itv)
 {
+	/*
+	 * MsgPack extensions have length greater or equal than 1 by
+	 * specification.
+	 */
+	assert(len > 0);
+
+	const char *end = *data + len;
+	uint32_t count = mp_load_u8(data);
+	len -= sizeof(uint8_t);
+
 	memset(itv, 0, sizeof(*itv));
-	uint32_t len = mp_load_u8(data);
-	for (uint32_t i = 0; i < len; ++i) {
+	for (uint32_t i = 0; i < count; ++i) {
+		if (len < 2)
+			return NULL;
 		uint32_t field = mp_load_u8(data);
 		int32_t value;
+		enum mp_type type = mp_typeof(**data);
+		if (type == MP_UINT) {
+			if (mp_check_uint(*data, end) > 0)
```

Comment:
The value is obtained using `mp_read_int32()` which checks the type. Do you really need extra validation for MP_INT and MP_UINT?

Reply:
> The value is obtained using mp_read_int32() which checks the type.

No it doesn't it returns `-1` if the MsgPack is not an integer https://github.com/tarantool/msgpuck/blob/28a7421cf7fa538a0180c79bd9c12ee0dd8c12eb/msgpuck.h#L2275-L2277

Reply:
What about this swith? I believe it checks type of the value: https://github.com/tarantool/msgpuck/blob/28a7421cf7fa538a0180c79bd9c12ee0dd8c12eb/msgpuck.h#L2253

Reply:
So, what does it do when the type is not an integer type?..

Reply:
I believe you can just return `itv` if `count` == 0. I think this looks better than an unnecessary check of two conditions.

Reply:
I see. True, I was wrong.